File: zigar-runtime/test/doc-examples/print-enum.zig
// print-enum.zig
const std = @import("std");

pub const Pet = enum {
    Dog,
    Cat,
    Snake,
    Chicken,
};

pub fn printTag(tag: Pet) void {
    std.debug.print("{s}: {d}\n", .{ @tagName(tag), @intFromEnum(tag) });
}

File: zigar-runtime/test/doc-examples/print-enum.js
// print-enum.js
import { printTag } from './print-enum.zig';

printTag('Dog');
printTag('Chicken');
try {
  printTag('Cow');
} catch (err) {
  console.error(err);
}

// console output:
// Dog: 0
// Chicken: 3
// TypeError: Enum item of the type Pet expected, received Cow
